Description: Fixes two errors causing FTBFS with gcc-14
 channelhopper.c:48:26: error: initialization of 'pthread_t'
 {aka 'long unsigned int'} from 'void *' makes integer from pointer without
 a cast [-Wint-conversion]
 .
 poc.c:516:17: error: 'return' with no value, in function returning non-void
 [-Wreturn-mismatch]
 .
 Backporting some changes applied upstream (Closes: #1081054).
Origin: backport, https://github.com/aircrack-ng/mdk4/commit/36ca143a2e6c0b75b5ec60143b0c5eddd3d2970c
Author: Sven Geuer <sge@debian.org>
Last-Update: 2024-10-04
--- a/src/channelhopper.c
+++ b/src/channelhopper.c
@@ -45,7 +45,7 @@
 int chans_out [MAX_CHAN_COUNT] = {0};
 
 pthread_t *hopper = NULL;
-pthread_t chan_sniffer = NULL;
+pthread_t chan_sniffer = (int)NULL;
 int hopper_useconds = 0;
 volatile int sniff = 0;
 pthread_mutex_t chan_thread_mutex;
--- a/src/attacks/poc.c
+++ b/src/attacks/poc.c
@@ -513,7 +513,7 @@
 	unsigned char s1, s2;
 
 	if(pascii == NULL || phex == NULL || len == 0)
-		return;
+		return (int)NULL;
 
 	str_len = strlen(pascii)/4;
 	if(str_len)
